connect/runtime/src/main/java/org/apache/kafka/connect/runtime/isolation/Plugins.java

```diff
@@ -249,20 +251,21 @@ public Converter newConverter(AbstractConfig config, String classPropertyName, C
      * @throws ConnectException if the {@link HeaderConverter} implementation class could not be found
      */
     public HeaderConverter newHeaderConverter(AbstractConfig config, String classPropertyName, ClassLoaderUsage classLoaderUsage) {
-        if (!config.originals().containsKey(classPropertyName)) {
-            // This configuration does not define the header converter via the specified property name
-            return null;
-        }
         HeaderConverter plugin = null;
         switch (classLoaderUsage) {
             case CURRENT_CLASSLOADER:
+                if (!config.originals().containsKey(classPropertyName)) {
```

Comment:
might want some commentary here, or in the javadoc, about how this works. the `CURRENT_CLASSLOADER` name isn't really clear within this limited context to understand why it makes sense to skip in this case (connector overrides probably only make sense if it's explicit in the connector config) since the `CURRENT_CLASSLOADER` doesn't make it clear that it is the connector's classloader.

Reply:
In fact, now I am wondering if we should rename the enums to be clearer. We can follow up on that separately, but I realize these branches aren't really very clear currently.

Reply:
I'm concerned about doing that now. If you're okay with waiting, I can file a separate issue for later.